relay-general/src/store/normalize.rs

```diff
@@ -1005,6 +1018,68 @@ impl<'a> Processor for NormalizeProcessor<'a> {
     }
 }
 
+/// If the logger is longer than [`MaxChars::Logger`], it returns a String with
+/// a shortened version of the logger. If not, the same logger is returned as a
+/// String.
+///
+/// To shorten the logger, all extra chars that don't fit into the maximum limit
+/// are removed, from the beginning of the logger.  Then, if the remaining
+/// substring contains a `.` somewhere but in the end, all chars until `.`
+/// (exclusive) are removed.
+///
+/// Additionally, the new logger is prefixed with `*`, to indicate it was
+/// shortened.
+fn shorten_logger(logger: &str) -> String {
+    let logger_len = bytecount::num_chars(logger.as_bytes());
+    if logger_len <= MaxChars::Logger.limit() {
+        return logger.to_owned();
+    }
+
+    let mut tokens = logger.split("").collect_vec();
+    // Remove empty str tokens from the beginning and end.
+    tokens.pop();
+    tokens.reverse(); // Prioritize chars from the end of the string.
+    tokens.pop();
+
+    remove_logger_extra_chars(&mut tokens);
+    remove_logger_word(&mut tokens);
+
+    tokens.reverse();
+    "*".to_owned() + &tokens.iter().join("")
```

Comment:
```suggestion
    tokens.join("*")
```

Reply:
`join` is the contrary operation to `split`. We only want the `*` at the beginning to indicate the logger was trimmed, not between each token (where a token is a character).

Reply:
Whoops, I guess I meant something like
```suggestion
    format!("*{}", tokens.join(""))
```